### DISCUSSION TITLE
Stop code between client.fetch_tan() und client.activate_session()

### DISCUSSION THREAD

**Discussion by jazzmoe** [alex-kn/comdirect-api-simple#11] [General]
Hi,

I am very new to Python and try to write my first scripts accessing my bank accounts. Unfortunately, after I run client.fetch_tan(user, password) the code runs further to client.activate_session() and creates an error. This is because in between I have to activate my photo tan. 

Is there any option to let the code "wait" until the tan is activated and then run the statement client.activate_session()?

```
client.fetch_tan(user, password)

# Here some condition would make sense to run the next line only when the TAN is activated

client.activate_session()
```

Thanks for you help,
Moritz

**Comment by alex-kn** [comment]
Hi Moritz,

Unfortunately, there is no way for the code to know when the TAN is activated. That is why you need to run `activate_session()` specifically after the manual TAN activation. There are a few ways you could achieve this, for example:

- Pause the script using a breakpoint in your IDE (probably only useful for development)
- Have a GUI or a server with different buttons/endpoints, one for fetching the TAN, and others for activating the session and doing other things. Note that `activate_session()` needs to be called on the same `client`-Object `fetch_tan()` was called on, because the token is saved there.

It is a bit laborious, but as far as I am aware of the comdirect API doesn't support something like 'wait for TAN activation, then continue' (at least the documentation doesn't say so). Would be a nice feature tho, I agree.

Hope I could be of some help :)
Alex

**Comment by georgoswonkos** [comment]
I just make my script wait for a keypress:

```
        print("Now activate your photo-TAN!")
        print("And then press <Enter> on the console.")
        input()
```
Regards,

Georg
